src/reformat/rgb_impl.rs

```diff
@@ -210,31 +210,39 @@ fn compute_rgb(y: f32, cb: f32, cr: f32, has_color: bool, mode: Mode) -> (f32, f
     )
 }
 
-fn clamped_pixel(
-    depth: u8,
-    // Technically, these two are options since one of them will always be Ok.
-    row: &AvifResult<&[u8]>,
-    row16: &AvifResult<&[u16]>,
-    index: usize,
-    max_channel: u16,
-) -> u16 {
-    if depth == 8 {
-        row.unwrap()[index] as u16
-    } else {
-        min(max_channel, row16.unwrap()[index])
+enum Row<'a> {
+    Absent,
+    Depth8(&'a [u8]),
+    Depth16(&'a [u16]),
+}
+
+impl image::Image {
+    // TODO: Move it to image.rs?
+    fn row_generic(&self, plane: Plane, row: u32) -> Row {
```

Comment:
optional: for later clean up. we can probably rename this function `row` and rename `row` to `row8`.

Reply:
Sure, can be done in another PR.